packages/keymaps/src/browser/keymaps-service.ts

```diff
@@ -128,62 +128,62 @@ export class KeymapsService {
      */
     async setKeybinding(newKeybinding: Keybinding, oldKeybinding: ScopedKeybinding | undefined): Promise<void> {
         return this.updateKeymap(() => {
-            let newAdded = false;
-            let isOldKeybindingDisabled = false;
-            let addedDisabledEntry = false;
-            const keybindings = [];
-            for (let keybinding of this.keybindingRegistry.getKeybindingsByScope(KeybindingScope.USER)) {
-                // search for the old keybinding and modify it
-                if (oldKeybinding && Keybinding.equals(keybinding, oldKeybinding, false, true)) {
-                    newAdded = true;
-                    keybinding = {
-                        ...keybinding,
-                        keybinding: newKeybinding.keybinding
-                    };
-                }
-
-                // we have an disabled entry for the same command and the oldKeybinding
-                if (oldKeybinding?.keybinding &&
-                    Keybinding.equals(keybinding, { ...newKeybinding, keybinding: oldKeybinding.keybinding, command: '-' + newKeybinding.command }, false, true)) {
-                    isOldKeybindingDisabled = true;
-                }
-                keybindings.push(keybinding);
-            }
-            if (!newAdded) {
-                keybindings.push({
-                    command: newKeybinding.command,
-                    keybinding: newKeybinding.keybinding,
-                    context: newKeybinding.context,
-                    when: newKeybinding.when,
-                    args: newKeybinding.args
-                });
-                newAdded = true;
+            const keybindings: Keybinding[] = [...this.keybindingRegistry.getKeybindingsByScope(KeybindingScope.USER)];
```

Comment:
The new code looks considerably shorter. Could you describe the change and convince me we're not missing any corner cases from the old code?